### PR TITLE
docs: refactor theme page to a single section

### PR DESCRIPTION
**Pull request by austinhung81**
Refactor theme page to a single section, and add some examples for each them page.
Referenced from the Material-UI

Screen Shot 2020-03-11 at 4 20 22 PM

### CODE REVIEW

File: packages/styled-docs/pages/breakpoints.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+> For example, you can write `<Box fontSize={["xs", "sm"]}/>` to make the font size responsive.
+
+<ThemeParser theme="breakpoints" />
+
+```jsx
+// responsive font size
+<Box fontSize={["xs", "sm", "md", "lg"]}>Trend Micro</Box>
+```
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/breakpoints.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/colors.mdx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  <ColorPalettes color="cyan" />
+</ColorWrapper>
+
+### Gray
+
+<ColorWrapper>
+  <ColorPalettes color="gray" />
+</ColorWrapper>
+
+### Black Transparent
```
Comment (cheton) [inline packages/styled-docs/pages/colors.mdx]: Use Sentence case for H3, H4, H5, H6

File: packages/styled-docs/pages/colors.mdx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+<ColorWrapper>
+  <ColorPalettes color="black" />
+</ColorWrapper>
+
+### White Transparent
```
Comment (cheton) [inline packages/styled-docs/pages/colors.mdx]: Use Sentence case for H3, H4, H5, H6

File: packages/styled-docs/pages/colors.mdx (hunk 2 above)
Comment (cheton) [reply packages/styled-docs/pages/colors.mdx]: How about using `### Black text` ?

File: packages/styled-docs/pages/colors.mdx (hunk 3 above)
Comment (cheton) [reply packages/styled-docs/pages/colors.mdx]: How about using `### White text`?

File: packages/styled-docs/pages/positions.mdx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+<ThemeParser theme="zIndices" />
+
+```jsx
+<Box position="relative" height="32x">
+  <Box position="absolute" zIndex="popover" left="4x" p="4x" bg="gray:80">z-index=popover</Box>
+  <Box position="absolute" zIndex="toast" top="9x" p="4x" bg="gray:70">z-index=toast</Box>
+</Box>
+```
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/positions.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/shadows.mdx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      <Box width="40x" height="24x" p="2x" boxShadow={`${colorMode}.md`}>boxShadow=md</Box>
+      <Box width="40x" height="24x" p="2x" boxShadow={`${colorMode}.lg`}>boxShadow=lg</Box>
+    </Stack>
+  );
+}
+
+render(<Example />);
+```
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/shadows.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/sizes.mdx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+```jsx
+<Stack direction="row" spacing=".5rem">
+  <Box width="40x" height="8x" p="1x" fontSize="sm" bg="gray:70">Height 2rem(32px)</Box>
+  <Box width="40x" height="16x" p="1x" fontSize="sm" bg="gray:70">Height 4rem(64px)</Box>
+  <Box width="40x" height="24x" p="1x" fontSize="sm" bg="gray:70">Height 6rem(96px)</Box>
+  <Box width="40x" height="32x" p="1x" fontSize="sm" bg="gray:70">Height 8rem(128px)</Box>
+</Stack>
+```
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/sizes.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/typography.mdx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+```jsx
+<Stack direction="column" spacing=".5rem">
+  <Box display="inline-block" pl="1x" lineHeight="sm" bg="green:80">Line height 1.25rem</Box>
+  <Box display="inline-block" pl="1x" lineHeight="lg" bg="green:80">Line height 1.5rem</Box>
+  <Box display="inline-block" pl="1x" lineHeight="2xl" bg="green:80">Line height 2rem</Box>
+</Stack>
+```
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/typography.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/spacing.mdx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    <Box display="inline-block" size="10x" bg="gray:70"/> 
+  </div>
+  <div>
+    <Box display="inline-block" mr="3x" size="10x" bg="gray:70"/> 
+    <Box display="inline-block" size="10x" bg="gray:70"/> 
+  </div>
+</>
+```
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/spacing.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/colors.mdx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+<ColorWrapper>
+  <ColorPalettes color="white" />
+</ColorWrapper>
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/colors.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/borders.mdx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+```jsx
+<Stack direction="row" spacing=".5rem">
+  <Box border={1} borderRadius="circle" size="12x" bg="gray:70"/> 
+  <Box border={1} borderRadius="sm" size="12x" bg="gray:70"/> 
+  <Box border={1} borderRadius="md" size="12x" bg="gray:70"/> 
+  <Box border={1} borderRadius="lg" size="12x" bg="gray:70"/> 
+</Stack>
+```
+
+## Configuration reference
```
Comment (cheton) [inline packages/styled-docs/pages/borders.mdx]: Use Title Case for H1, H2

File: packages/styled-docs/pages/colors.mdx [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+### Green
+
+<ColorWrapper>
+  <ColorPalettes color="green" />
+</ColorWrapper>
+
+### Teal
+
+<ColorWrapper>
+  <ColorPalettes color="teal" />
```
Comment (tinaClin) [inline packages/styled-docs/pages/colors.mdx:63]: 突然發現有顏色不對
zeplin上的
image

docs上的
image

File: packages/styled-docs/pages/spacing.mdx [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+`margin`, and `top`, `left`, `right`, `bottom` styles.
+
+<ThemeParser theme="space" />
+
+By default, the values are proportional, so 1 spacing unit is equal to
+`0.25rem`, which translates to `4px` by default in common browsers.
+
+| Name | Space   | Pixels |                                    |
+| ---- | ------- | ------ | ---------------------------------- |
+| 0    | 0       | 0px    | <Box bg="blue:60" h="4x" w={0}/>   |
```
Comment (tinaClin) [inline packages/styled-docs/pages/spacing.mdx:17]: 最後一欄沒有效果
image

File: packages/styled-docs/pages/colors.mdx (hunk 11 above)
Comment (cheton) [reply packages/styled-docs/pages/colors.mdx:63]: This will be fixed in this PR

**Review by cheton — CHANGES_REQUESTED** [review]

File: packages/styled-docs/pages/spacing.mdx (hunk 12 above)
Comment (tinaClin) [reply packages/styled-docs/pages/spacing.mdx:17]: 啊...又出現了...
image

File: packages/styled-docs/pages/colors.mdx (hunk 2 above)
Comment (austinhung81) [reply packages/styled-docs/pages/colors.mdx]: > How about using `### Black text` ?

But it's talk about color, not specific for text. I'll think long and hard.

Comment (cheton) [reply packages/styled-docs/pages/colors.mdx]: Simply “Black” is fine if we don’t have a better naming at the moment.

**Review by cheton — APPROVED** [review]

**Review by cheton — APPROVED** [review]

File: packages/styled-docs/pages/colors.mdx (hunk 2 above)
Comment (austinhung81) [reply packages/styled-docs/pages/colors.mdx]: Good suggestion, thanks!

**Review by tinaClin — APPROVED** [review]

**Pull request merged**